src/viam/examples/modules/simple/main.cpp

```diff
@@ -24,118 +23,126 @@
 using viam::component::generic::v1::GenericService;
 using namespace viam::sdk;
 
-class MyModule : public GenericService::Service, public Component {
+// Printer is a modular resource that can print a to_print value to STDOUT when
+// a DoCommand request is received or when reconfiguring. The to_print value
+// must be provided as an attribute in the config.
+class Printer : public GenericService::Service, public Component {
```

Comment:
from speaking with @abe-winter, I think this should properly inherit from `Generic`, not `GenericService::Service`. Seems like a bit more of a refactor though.

Reply:
Good point. Switched it to inherit from only `Generic`. Required some refactoring, but things seem simpler now.